module-2/backend/week_two.md

```diff
@@ -4,38 +4,90 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
+4. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.<br>
+*These are class methods.<br>
+Cars and dealerships would have a one to many relationship in that a car would belong_to a dealership but a dealership could has_many cars.*
+
+5. What do they allow you to do?<br>
+*They allow you to associate tables with other tables via a foriegn key.*
+
+6. What's the difference between agile workflow and waterfall method?<br>
+*Agile workflow aims to break up a project into smaller parts that are tested and complete on their own. You could therefore have a very small working part of your project before your move on to starting the next part.   Waterfall method takes the whole project together, plans for it and impltements it.  This makes it much harder to make changes along the way and adapt to new problems, which would be much easier in the agile method.*
+
+7. What is the difference between `#new` and `#create`?<br>
+*Create combines '#new' and '#save', so it makes a new instance of your object and it saves it into your database. '#new' just creates a new instance without saving it.*
+
+
+8. At a basic level, what does cURL allow you to do?<br>
+*cURL allows you to transfer data to a server through the command line.*
+
+9. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.<br>
+*They have a many to many relationship, since students have many teachers and teachers also have many students. Therefore we would have to make a join table, called students_teachers. This students_teachers would have it's own primary id, as well as the teacher_id and the student_id. In the schema diagram, these last two ids on the students_teachers table would point to the primary ids on the student table and teacher table, respectively.*
+
+
+10. Define foreign key, primary key, and schema.<br>
+*A primary key is the unique identifier for the row of data in that table. A foreign key is a primary key in a different table, that is used to make an association in our current table. Schema is the structure of our database, showing us what is in each table.*
+
+11. Describe the relationship between a foreign key on one table and a primary key on another table. <br>
+*A primary key of one table becomes the foreign key on a different associated table.  On the car / dealerships example, where the dealership has many cars, the car table would have the foreign key of the dealership on it. So car table would have id, name, make, model, etc, and dealership_id.*
+
+12. What are the parts of an HTTP response?<br>
+*A status line, a header, and a body*
+
+13. `Rack::Test` allows us to test our controllers in isolation. What are some of the methods it gives us to simulate the request/response cycle?<br>
+*Methods include: .to eq(), expect(page), click_on(link), to_not eq()*
```

Comment:
These are actually RSpec and Capybara methods. We didn't teach Rack::Test this time around, so this is totally on us. Should have probably taken this question out.